drivers/gles3/storage/config.cpp

```diff
@@ -169,8 +169,31 @@ Config::Config() {
 
 	//Adreno 3xx Compatibility
 	const String rendering_device_name = String::utf8((const char *)glGetString(GL_RENDERER));
-	//TODO: Check the number between 300 and 399(?)
-	adreno_3xx_compatibility = (rendering_device_name.left(13) == "Adreno (TM) 3");
+	if (rendering_device_name.left(13) == "Adreno (TM) 3") {
+		adreno_3xx_compatibility = true;
+
+		// ignore driver version 331+
+		const String gl_version = String::utf8((const char *)glGetString(GL_VERSION));
+		// Adreno 3xx examples (https://opengles.gpuinfo.org/listreports.php):
+		// ===========================================================================
+		// OpenGL ES 3.0 V@84.0 AU@ (CL@)
+		// OpenGL ES 3.0 V@127.0 AU@ (GIT@I96aee987eb)
+		// OpenGL ES 3.0 V@140.0 AU@ (GIT@Ifd751822f5)
+		// OpenGL ES 3.0 V@251.0 AU@08.00.00.312.030 (GIT@Ie4790512f3)
+		// OpenGL ES 3.0 V@269.0 AU@ (GIT@I109c45a694)
+		// OpenGL ES 3.0 V@331.0 (GIT@35e467f, Ice9844a736) (Date:04/15/19)
+		// OpenGL ES 3.0 V@415.0 (GIT@d39f783, I79de86aa2c, 1591296226) (Date:06/04/20)
+		// OpenGL ES 3.0 V@0502.0 (GIT@09fef447e8, I1fe547a144, 1661493934) (Date:08/25/22)
+		int start_pos = gl_version.find("V@");
+		if (start_pos > 0) {
+			start_pos += 2;
+			int end_pos = gl_version.find(" ", start_pos);
+			String driver_version = gl_version.substr(start_pos, end_pos - start_pos);
+			if (driver_version.is_valid_float() && driver_version.to_float() >= 331.0) {
+				adreno_3xx_compatibility = false;
+			}
+		}
+	}
```

Comment:
Personally, I'd have done `gl_version.get_slice("V@", 1).get_slice(".", 0)` to get the number in a single line. It'll still have a leading zero though.

Reply:
looks good

```cpp
String driver_version = gl_version.get_slice("V@", 1).get_slice(" ", 0);
```
**INPUT:** `OpenGL ES 3.0 V@0502.0 (GIT@09fef447e8`...
print_line **OUTPUT:** `driver_version:  0502.0 to_float:  502`